src/scene/forward-renderer.js

```diff
@@ -1179,7 +1237,7 @@ Object.assign(ForwardRenderer.prototype, {
             light = lights[i];
             type = light._type;
             if (light.castShadows && light.enabled && light.shadowUpdateMode !== SHADOWUPDATE_NONE) {
-                if (type !== LIGHTTYPE_DIRECTIONAL) {
+                if (type !== LIGHTTYPE_DIRECTIONAL && type !== LIGHTTYPE_AREA) {
```

Comment:
It will probably be good to implement frustum culling for area lights as well, otherwise not many local area lights will be able to be placed in the scene as they will all need to be "used".

Do these lights have range, similar to point / spot lights? If yes, perhaps area light culling could be based on spot light with range and 180deg cone?

Reply:
There is no implicit attenuation or falloff calculated, but it does seem like a good idea to cap it. A couple of things to consider; area lights can be different shapes (disk, line etc), so any culling would need to accommodate this. I guess this can just be added on a case by case basis? Also, they also commonly allow double sided illumination which is inexpensive to calculate, so in the quad instance the hemisphere would become a sphere

Reply:
That sounds good. The culling function calls light.getBoundingSphere where it handles spot and point light differently, we'd just need to add some handling of area light per shape type. For now either half sphere or sphere should be returned as you described.